proxy/src/telemetry/metrics/mod.rs

```diff
@@ -427,56 +427,54 @@ impl Aggregate {
             },
 
             Event::StreamRequestFail(ref req, _) => {
-                let labels = Arc::new(RequestLabels::new(req));
                 self.update(|metrics| {
-                    metrics.request_total(&labels).incr();
+                    metrics.request_total(RequestLabels::new(req)).incr();
                 })
             },
 
             Event::StreamRequestEnd(ref req, _) => {
-                let labels = Arc::new(RequestLabels::new(req));
                 self.update(|metrics| {
-                    metrics.request_total(&labels).incr();
+                    metrics.request_total(RequestLabels::new(req)).incr();
                 })
             },
 
             Event::StreamResponseEnd(ref res, ref end) => {
-                let labels = Arc::new(ResponseLabels::new(
-                    res,
-                    end.grpc_status,
-                ));
                 self.update(|metrics| {
-                    metrics.response_total(&labels).incr();
-                    metrics.response_latency(&labels).add(end.since_request_open);
+                    let labels = ResponseLabels::new(
+                        res,
+                        end.grpc_status,
+                    );
+                    metrics.response_total(labels.clone()).incr();
+                    metrics.response_latency(labels).add(end.since_request_open);
                 });
             },
 
             Event::StreamResponseFail(ref res, ref fail) => {
-                // TODO: do we care about the failure's error code here?
-                let labels = Arc::new(ResponseLabels::fail(res));
                 self.update(|metrics| {
-                    metrics.response_total(&labels).incr();
-                    metrics.response_latency(&labels).add(fail.since_request_open);
+                    // TODO: do we care about the failure's error code here?
+                    let labels = ResponseLabels::fail(res);
```

Comment:
in other blocks, labels is constructed outside of update

Reply:
Only for transport events, since the transport labels are `Copy`. For request/response events, I'm constructing the labels inside the closure to avoid and additional clone (we can borrow the event, since it's a reference, but we can't move the labels into the closure).